### PR TITLE
fix(sampling):  Persist project stats data in sampling store, fixing concurrency  - [TET-347]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Problem:**

- There is a race condition in the `ServerSideSampling` component
- `projectStats` for 48h is fetched [here](https://github.com/getsentry/sentry/blob/master/static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx#L115-L121)
- but when clicking `Get Started`, it may still be `undefined` in the handler [here](https://github.com/getsentry/sentry/blob/master/static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx#L194-L216)
- this leads to the modal being in an infinite loading state because of this check [here](https://github.com/getsentry/sentry/blob/master/static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx#L113) so even if the 30d project stats load, the 48h is `undefined` from the parent component, and even once it loads, it never updates

**Solution:**

- Fetch project stats from the last 30d and 48d and save the responses in the `ServerSideSamplingStore`

### CODE REVIEW

**Review by matejminar — APPROVED** [review]
Looks good to me.

File: static/app/actionCreators/serverSideSampling.tsx [hunk 1]
```diff
@@ -14,12 +14,13 @@ export function fetchSamplingSdkVersions({
   orgSlug: Organization['slug'];
   projectID: Project['id'];
 }): Promise<SamplingSdkVersion[]> {
-  const {samplingDistribution} = ServerSideSamplingStore.getState();
-  const {startTimestamp, endTimestamp, project_breakdown} = samplingDistribution;
+  const {distribution} = ServerSideSamplingStore.getState();
+  const {startTimestamp, endTimestamp, project_breakdown} = distribution.data ?? {};
+
+  ServerSideSamplingStore.fetchSdkVersions();
```
Comment (Zylphrex) [inline static/app/actionCreators/serverSideSampling.tsx:20]: This is a nit but the name of all these `fetch*()` functions on the store seems to imply that they'll do the fetching but in reality, they just mark the start of the fetch and then you have to explicitly fetch and load the data into the store. Would be nice if they had clearer names.

**Review by Zylphrex — APPROVED** [review]
Thanks for addressing this so quickly! 😄

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 2]
```diff
@@ -95,7 +97,7 @@ export function ServerSideSampling({project}: Props) {
       return;
     }
 
-    async function fetchRecommendedSdkUpgrades() {
+    async function fetchData() {
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:100]: one depends on another... but I can make the last two run parallel... thanks

File: static/app/actionCreators/serverSideSampling.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/actionCreators/serverSideSampling.tsx:20]: sure thing... will add a start fetching or something 😉

**Comment by priscilawebdev** [comment]
@Zylphrex will address the suggestions in a follow-up 😉

**Pull request merged**